File: app/src/main/java/com/productivity/habits/ui/daily/HistoricalBanner.kt
package com.productivity.habits.ui.daily

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.expandVertically
import androidx.compose.animation.shrinkVertically
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.History
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import java.time.LocalDate
import java.time.format.DateTimeFormatter

@Composable
fun HistoricalBanner(
    selectedDate: LocalDate,
    onReturnToToday: () -> Unit,
    modifier: Modifier = Modifier
) {
    val isToday = selectedDate == LocalDate.now()
    val isPast = selectedDate.isBefore(LocalDate.now())

    AnimatedVisibility(
        visible = !isToday,
        enter = expandVertically(),
        exit = shrinkVertically()
    ) {
        val formattedDate = selectedDate.format(DateTimeFormatter.ofPattern("EEE, MMM d, yyyy"))
        val label = if (isPast) "Viewing Past: $formattedDate" else "Viewing Future: $formattedDate"

        Row(
            modifier = modifier
                .fillMaxWidth()
                .background(MaterialTheme.colorScheme.tertiaryContainer.copy(alpha = 0.85f))
                .padding(horizontal = 16.dp, vertical = 6.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier.weight(1f)
            ) {
                Icon(
                    imageVector = Icons.Default.History,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onTertiaryContainer,
                    modifier = Modifier.size(18.dp)
                )
                Text(
                    text = label,
                    style = MaterialTheme.typography.bodySmall,
                    fontWeight = FontWeight.Medium,
                    color = MaterialTheme.colorScheme.onTertiaryContainer,
                    modifier = Modifier.padding(start = 8.dp)
                )
            }

            TextButton(
                onClick = onReturnToToday,
                colors = ButtonDefaults.textButtonColors(
                    contentColor = MaterialTheme.colorScheme.onTertiaryContainer
                )
            ) {
                Text(
                    text = "Return to Today",
                    style = MaterialTheme.typography.labelMedium,
                    fontWeight = FontWeight.Bold
                )
            }
        }
    }
}

File: lib/ui/daily/widgets/historical_banner.dart
import 'package:flutter/material.dart';
import 'package:flutter/widget_previews.dart';
import 'package:intl/intl.dart';
import '../../common/previews/phial_previews.dart';

class HistoricalBanner extends StatelessWidget {
  final DateTime selectedDate;
  final VoidCallback onReturnToToday;

  const HistoricalBanner({
    super.key,
    required this.selectedDate,
    required this.onReturnToToday,
  });

  @override
  Widget build(BuildContext context) {
    final now = DateTime.now();
    final today = DateTime(now.year, now.month, now.day);
    final currentSelected = DateTime(
      selectedDate.year,
      selectedDate.month,
      selectedDate.day,
    );

    final isToday = currentSelected == today;
    final isPast = currentSelected.isBefore(today);

    if (isToday) {
      return const SizedBox.shrink();
    }

    final theme = Theme.of(context);
    final formattedDate =
        DateFormat('EEE, MMM d, yyyy').format(selectedDate);
    final label = isPast
        ? 'Viewing Past: $formattedDate'
        : 'Viewing Future: $formattedDate';

    return Container(
      width: double.infinity,
      color: theme.colorScheme.tertiaryContainer.withValues(alpha: 0.85),
      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 6),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Expanded(
            child: Row(
              children: [
                Icon(
                  Icons.history,
                  size: 18,
                  color: theme.colorScheme.onTertiaryContainer,
                ),
                const SizedBox(width: 8),
                Flexible(
                  child: Text(
                    label,
                    style: theme.textTheme.bodySmall?.copyWith(
                      fontWeight: FontWeight.w500,
                      color: theme.colorScheme.onTertiaryContainer,
                    ),
                    maxLines: 1,
                    overflow: TextOverflow.ellipsis,
                  ),
                ),
              ],
            ),
          ),
          TextButton(
            onPressed: onReturnToToday,
            style: TextButton.styleFrom(
              foregroundColor: theme.colorScheme.onTertiaryContainer,
              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
              minimumSize: Size.zero,
              tapTargetSize: MaterialTapTargetSize.shrinkWrap,
            ),
            child: Text(
              'Return to Today',
              style: theme.textTheme.labelMedium?.copyWith(
                fontWeight: FontWeight.bold,
                color: theme.colorScheme.onTertiaryContainer,
              ),
            ),
          ),
        ],
      ),
    );
  }
}

// ==========================================
// Widget Previews
// ==========================================

@PhialMultiBrightnessPreview(name: 'Historical Banner - Past Date', group: 'Daily')
Widget previewHistoricalBannerPast() {
  return PhialPreviewWrapper(
    padding: EdgeInsets.zero,
    child: HistoricalBanner(
      selectedDate: DateTime.now().subtract(const Duration(days: 3)),
      onReturnToToday: () {},
    ),
  );
}

@Preview(name: 'Historical Banner - Future Date', group: 'Daily')
Widget previewHistoricalBannerFuture() {
  return PhialPreviewWrapper(
    padding: EdgeInsets.zero,
    child: HistoricalBanner(
      selectedDate: DateTime.now().add(const Duration(days: 2)),
      onReturnToToday: () {},
    ),
  );
}
